release/tar/linux/opensearch-dashboards-tar-build.sh

```diff
@@ -0,0 +1,153 @@
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+
+#
+# About:        This shell script generate an opensearch-dashboards tarball with all the provided plugins installed and ready to use.
+#               It will create a folder named target that will consist of the tarball and shasum of the same.
+#               A manifest file and right permissions set up is essential to download the required artifacts
+# Dependencies: yq (More info: https://github.com/mikefarah/yq/tree/v4.4.1#install)
+#               shasum
+# Usage:        ./opensearch-dashboards-tar-build.sh -f <path/to/manifest/file/>
+#
+
+set -e
+
+function usage() {
+    echo "Usage: $0 [args]"
+    echo -e "Supported manifest schema-version: 1.0"
+    echo ""
+    echo "Arguments:"
+    echo -e "-f MANIFEST_FILE\t[Required] Path/to/manifest.yml file"
+    echo -e "-h help"
+}
+
+while getopts ":hf:" arg; do
+    case $arg in
+        h)
+            usage
+            exit 1
+            ;;
+        f)
+            MANIFEST_FILE=$OPTARG
+            ;;
+        :)
+            echo "Error: -${OPTARG} requires an argument"
+            usage
+            exit 1
+            ;;
+        ?)
+            echo "Invalid option: -${arg}"
+            exit 1
+            ;;
+    esac
+done
+
+if [ -z "$MANIFEST_FILE" ]; then
+    echo "error: You must specify the manifest file"
+    usage
+    exit 1
+fi
+
+if [ ! -f "$MANIFEST_FILE" ]; then
+    echo "error: The given file doesnot exist. Please check the specified path"
+    usage
+    exit 1
+fi
+
+SCHEMA_VERSION=`yq eval '.manifest.schema-version' $MANIFEST_FILE`
+PLATFORM=`yq eval '.build.platform' $MANIFEST_FILE`
+ARCHITECTURE=`yq eval '.build.architecture' $MANIFEST_FILE`
+VERSION=`yq eval '.build.version' $MANIFEST_FILE`
+UNDERLYING_ARCH=`uname -p`
+
+if [ -z "$VERSION" ]; then
+    echo "error: Please specify the version in the manifest file"
+    usage
+    exit 1
+fi
+
+if [ "$SCHEMA_VERSION" != "1.0" ]; then
+    echo "error: This script only supports manifest schema version 1.0. Please use the manifest file with right schema version"
+    usage
+    exit 1
+fi
+
+if [ "$PLATFORM" != "linux" ]; then
+    echo "error: $PLATFORM platform is not supported yet! Only supported platform is linux"
+    usage
+    exit 1
+fi
+
+if [ "$ARCHITECTURE" != "x64" ] && [ "$ARCHITECTURE" != "arm64" ]
+then
+    echo "error: $ARCHITECTURE architecture is not supported yet! Only supported architectures are x64 and arm64"
+    usage
+    exit 1
+fi
+
+if [[ ( "$ARCHITECTURE" == "arm64" && ( "$UNDERLYING_ARCH" != "aarch64" && "$UNDERLYING_ARCH" != "arm64" )) ||  ( "$ARCHITECTURE" == "x64" && ( "$UNDERLYING_ARCH" != "x86_64" && "$UNDERLYING_ARCH" != "amd64" )) ]]
```

Comment:
Why don't we just figure out ARCHITECTURE from UNDERLYING_ARCH? Then the user doesn't need to specify the architecture. If you want x86, run this script on an x86 box. If you want arm, run it on arm.

What you have is fine for now; this is a suggestion for future improvements.

Reply:
Noted as an enhancement